### PR TITLE
Feature/message broker publisher

### PR DESCRIPTION
**Pull request by jkoenig134**

### CODE REVIEW

**Comment by jkoenig134** [comment]
@sebbi08 added you as this could also be interesting for you.

**Comment by Milena-Czierlinski** [comment]
The Event Introduction will need adjustments, too.

**Comment by jkoenig134** [comment]
@Milena-Czierlinski! right - thanks, I forgot this. That's why I wanted to not talk about any modules there :/

**Comment by jkoenig134** [comment]
@Milena-Czierlinski! right - thanks, I forgot this. That's why I wanted to not talk about any modules there :/

File: _includes/scenarios/operate/configuration.md [hunk 1]
```diff
@@ -281,6 +284,108 @@ It is not recommended to use this Module for production scenarios.
...
+
+  - type `Redis`
+
+    **example**
+
+    ```jsonc
+    {
+      "type": "Redis",
+      "config": {
+        "url": "redis://example.com:6379"
```
Comment (sebbi08) [inline _includes/scenarios/operate/configuration.md:380]: Maybe we should add a sample for a Redis with authentication or link to the original Redis URL schema documentation

https://github.com/lettuce-io/lettuce-core/wiki/Redis-URI-and-connection-details

Same for the other Types, I don't know where it is supported and not

Comment (jkoenig134) [reply _includes/scenarios/operate/configuration.md:380]: Great idea, i'll add link to all URL schemes.

**Comment by jkoenig134** [comment]
Conflicts resolved, waiting for a review.

File: _docs_operate/configuration.md [hunk 2]
```diff
@@ -306,6 +309,115 @@ It is not recommended to use this Module for production scenarios.
...
+
+  - type `MQTT`
+
+    **example**
+
+    ```jsonc
+    {
+      "type": "MQTT",
+      "config": {
+        "brokerUrl": "mqtt://example.com:1883"
```
[the remark this answers is not in the document]
Comment (jkoenig134) [reply _docs_operate/configuration.md:370]: good catch, sadly not tackleable here.

[the remark this answers is not in the document]
Comment (jkoenig134) [reply _docs_operate/configuration.md:370]: I'll PR this in the connector repo

File: _docs_operate/configuration.md [hunk 3]
```diff
@@ -306,6 +309,115 @@ It is not recommended to use this Module for production scenarios.
...
+        "keyFile": "path/to/keyfile.json"
+      }
+    }
+    ```
+
+    **configuration**
+
+    - projectId `string, required` - the project id of the Google Cloud project
+    - topicName `string, required` - the name of the PubSub topic to publish to
+    - keyFile `string, required` - the location of the key file to authenticate with the Google Cloud project
```
Comment (slavistan) [inline _docs_operate/configuration.md:400]: If a file path can be relative, I'd like to know how this path will be resolved. The two options here are (1) relative to the working directory from which the connector is started and (2) relative to the directory of the configuration file we're writing.

Would be helpful to be explicit about this here.

Comment (jkoenig134) [reply _docs_operate/configuration.md]: 1. this is only routed to the google pubsub library, so we don't know and don't care about relative urls
2. this is in docker and we define nowhere were our base path is, so this shouldn't matter

wtb just saying they should use an absolut path?

Comment (jkoenig134) [reply _docs_operate/configuration.md]: > It's probably the working directory of the connector

and the user shouldn't care about this working directory.

> Sysadmins will hate you if you force them to use absolute paths

it's VERY common in the docker world that you have absolute paths. You mount it into the container using an absolute path, why should you care if you use an absolute path in the config?

**Pull request merged**